File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curr = prices[0];
        int maxi = 0;
        for(int i=1; i<prices.size(); i++){
            if (curr<prices[i]){
                maxi = max(maxi,prices[i]-curr);
            }else{
                curr = prices[i];
            }
        }
        return maxi;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr = prices[0]
        maxi = 0
        for i in range(len(prices)-1):
            if curr<prices[i+1]:
                maxi = max(maxi,prices[i+1]-curr)
            else:
                curr = prices[i+1]
        return maxi